[PUI] Add coverage testing

matmair opened the pull request:
This adds code coverage for the PUI tests and reporting to coveralls.

matmair commented:
@wolflu05 you are more experienced in typescript/frontend. Is anything sticking out with this implementation?

wolflu05 commented:
Sorry, I have not that much experience with frontend testing, never really done this before.

Regarding the changes on the vite config, does this mean we now ship sourcemaps with all builds? Is that wanted?

And the plugin you added, has that any influence on the build or dev environment?

matmair commented:
I have removed tests I added before, they lowered the coverage a lot and were not that great - still wrapping my head around writing efficient tests. The focus of this PR - coverage - works great, expanding the tests should probably be part of further PUI development.

wolflu05 commented on .github/workflows/qc_checks.yaml:
```diff
@@ -487,13 +488,30 @@ jobs:
           name: playwright-report
           path: src/frontend/playwright-report/
           retention-days: 30
+      - name: Report coverate
```
typo
```suggestion
      - name: Report coverage
```

matmair replied:
Thank you!

SchrodingersGat commented:
@matmair please address conflicts here

matmair commented:
@SchrodingersGat merge is in!
